internal/provider/resources/resource_cluster_workspace_authorization.go

```diff
@@ -0,0 +1,381 @@
+
+	// Read Terraform prior state data into the model
+	resp.Diagnostics.Append(req.State.Get(ctx, &data)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	var diags diag.Diagnostics
+	var updateClusterRequest platform.UpdateClusterRequest
+	updateDedicatedClusterRequest := platform.UpdateDedicatedClusterRequest{
+		WorkspaceIds: nil,
+	}
+
+	err := updateClusterRequest.FromUpdateDedicatedClusterRequest(updateDedicatedClusterRequest)
+	if err != nil {
+		tflog.Error(ctx, fmt.Sprintf("failed to update cluster error: %v", err))
+		resp.Diagnostics.AddError(
+			"Client Error",
+			fmt.Sprintf("Unable to update cluster request body, got error: %s", err),
+		)
+		return
+	}
+
+	cluster, err := r.platformClient.UpdateClusterWithResponse(ctx, r.organizationId, data.ClusterId.ValueString(), updateClusterRequest)
+	if err != nil {
+		tflog.Error(ctx, "failed to update cluster", map[string]interface{}{"error": err})
+		resp.Diagnostics.AddError(
+			"Client Error",
+			fmt.Sprintf("Unable to update cluster, got error: %s", err),
+		)
+		return
+	}
+	_, diagnostic := clients.NormalizeAPIError(ctx, cluster.HTTPResponse, cluster.Body)
+	if diagnostic != nil {
+		resp.Diagnostics.Append(diagnostic)
+		return
+	}
+
+	// Wait for the cluster to be updated (or fail)
+	stateConf := &retry.StateChangeConf{
+		Pending:    []string{string(platform.ClusterStatusCREATING), string(platform.ClusterStatusUPDATING)},
+		Target:     []string{string(platform.ClusterStatusCREATED), string(platform.ClusterStatusUPDATEFAILED), string(platform.ClusterStatusCREATEFAILED)},
+		Refresh:    r.resourceRefreshFunc(ctx, cluster.JSON200.Id),
+		Timeout:    3 * time.Hour,
+		MinTimeout: 1 * time.Minute,
+	}
+
+	// readyCluster is the final state of the cluster after it has reached a target status
+	readyCluster, err := stateConf.WaitForStateContext(ctx)
+	if err != nil {
+		resp.Diagnostics.AddError("Cluster update failed", err.Error())
+		return
+	}
+
+	diags = data.ReadFromResponse(readyCluster.(*platform.Cluster))
+	if diags.HasError() {
+		resp.Diagnostics.Append(diags...)
+		return
+	}
+
+	tflog.Trace(ctx, fmt.Sprintf("Deleted cluster workspace authorization for: %v", data.ClusterId.ValueString()))
+
+	// Save updated data into Terraform state
+	resp.Diagnostics.Append(resp.State.Set(ctx, &data)...)
+}
+
+// resourceRefreshFunc returns a retry.StateRefreshFunc that polls the platform API for the cluster status
+// If the cluster is not found, it returns "DELETED" status
+// If the cluster is found, it returns the cluster status
+// If there is an error, it returns the error
+// WaitForStateContext will keep polling until the target status is reached, the timeout is reached or an err is returned
+func (r *ClusterWorkspaceAuthorizationResource) resourceRefreshFunc(ctx context.Context, clusterId string) retry.StateRefreshFunc {
+	return func() (any, string, error) {
+		cluster, err := r.platformClient.GetClusterWithResponse(ctx, r.organizationId, clusterId)
```

Comment:
can we turn this into a helper with param r.platformClient, r.organizationId, clusterId since its also used in cluster_resource